pyiceberg/table/snapshots.py

```diff
@@ -116,3 +168,193 @@ class MetadataLogEntry(IcebergBaseModel):
+        elif data_file.content == DataFileContent.EQUALITY_DELETES:
+            self.removed_delete_files += 1
+            self.removed_eq_delete_files += 1
+            self.removed_eq_deletes += data_file.record_count
+        else:
+            raise ValueError(f"Unknown data file content: {data_file.content}")
+
+    def build(self) -> Dict[str, str]:
+        def set_when_positive(properties: Dict[str, str], num: int, property_name: str) -> None:
+            if num > 0:
+                properties[property_name] = str(num)
+
+        properties: Dict[str, str] = {}
+        set_when_positive(properties, self.added_size, ADDED_FILE_SIZE)
+        set_when_positive(properties, self.removed_size, REMOVED_FILE_SIZE)
+        set_when_positive(properties, self.added_files, ADDED_DATA_FILES)
+        set_when_positive(properties, self.removed_files, DELETED_DATA_FILES)
+        set_when_positive(properties, self.added_eq_delete_files, EQUALITY_DELETE_FILES)
+        set_when_positive(properties, self.removed_eq_delete_files, REMOVED_EQUALITY_DELETE_FILES)
+        set_when_positive(properties, self.added_pos_delete_files, ADDED_POSITION_DELETE_FILES)
+        set_when_positive(properties, self.removed_pos_delete_files, REMOVED_POSITION_DELETE_FILES)
+        set_when_positive(properties, self.added_delete_files, ADDED_DELETE_FILES)
+        set_when_positive(properties, self.removed_delete_files, REMOVED_DELETE_FILES)
+        set_when_positive(properties, self.added_records, ADDED_RECORDS)
+        set_when_positive(properties, self.deleted_records, DELETED_RECORDS)
+        set_when_positive(properties, self.added_pos_deletes, ADDED_POSITION_DELETES)
+        set_when_positive(properties, self.removed_pos_deletes, REMOVED_POSITION_DELETES)
+        set_when_positive(properties, self.added_eq_deletes, ADDED_EQUALITY_DELETES)
+        set_when_positive(properties, self.removed_eq_deletes, REMOVED_EQUALITY_DELETES)
+
+        return properties
+
+
+def _truncate_table_summary(summary: Summary, previous_summary: Mapping[str, str]) -> Summary:
+    for prop in {
+        TOTAL_DATA_FILES,
+        TOTAL_DELETE_FILES,
+        TOTAL_RECORDS,
+        TOTAL_FILE_SIZE,
+        TOTAL_POSITION_DELETES,
+        TOTAL_EQUALITY_DELETES,
+    }:
+        summary[prop] = '0'
+
+    if value := previous_summary.get(TOTAL_DATA_FILES):
+        summary[DELETED_DATA_FILES] = value
+    if value := previous_summary.get(TOTAL_DELETE_FILES):
+        summary[REMOVED_DELETE_FILES] = value
+    if value := previous_summary.get(TOTAL_RECORDS):
+        summary[DELETED_RECORDS] = value
+    if value := previous_summary.get(TOTAL_FILE_SIZE):
+        summary[REMOVED_FILE_SIZE] = value
+    if value := previous_summary.get(TOTAL_POSITION_DELETES):
+        summary[REMOVED_POSITION_DELETES] = value
+    if value := previous_summary.get(TOTAL_EQUALITY_DELETES):
+        summary[REMOVED_EQUALITY_DELETES] = value
+
+    return summary
+
+
+def _merge_snapshot_summaries(
```

Comment:
Is this really a merge? To me, a merge is like adding two summaries together. This is actually updating from a previous snapshot summary.

Reply:
Changed it to `_update_snapshot_summaries`